src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs

```diff
@@ -27,6 +27,10 @@ private BatchFixAllProvider()
         {
         }
 
+        public override IEnumerable<FixAllScope> GetSupportedFixAllScopes()
+            => ImmutableArray.Create(FixAllScope.Document, FixAllScope.Project,
+                FixAllScope.Solution, FixAllScope.ContainingMember, FixAllScope.ContainingType);
```

Comment:
oh... i see.  you're making it so that anyone that uses BatchFixAllProvider.Instance now is opted into this... hrmmmm

that's potentialy more concerning.  though i guess we can do this and then filter this out later when we process the diagnostics.  i'm not a huge fan though as i think it might be better to keep old behavior and have people opt in... 

but i do get that this is beneficial for all the C#/VB fixers out there that won't update hat could use this :-/

Reply:
> i see. you're making it so that anyone that uses BatchFixAllProvider.Instance now is opted into this

Yes, this is something Sam and I discussed and decided to change.

> that's potentialy more concerning

I don't think so. Batch fixer works by merging code actions for diagnostics to fixed, and if it works fine for all diagnostics in a document, it should work fine for subset of those in a document as well.

> i'm not a huge fan though as i think it might be better to keep old behavior and have people opt in...

There is no way to opt-in. The BatchFixer is a complete FixAllProvider implementation. We can potentially have a new BatchFixer instance that supports additional modes, but in reality that would mean none of the existing code fixers will ever get updated to support new scopes. I would agree with your concern if this was done for FixAll for code refactorings, as we do not have a good way to guarantee that if it works on a document, it will work on a span. But I think making this assumption for code fixers is fine.